api/java/org/openyolo/api/CredentialClient.java

```diff
@@ -68,29 +69,43 @@
 
     private static final String LOG_TAG = "CredentialClient";
 
-    private static final AtomicReference<CredentialClient> INSTANCE_REF
-            = new AtomicReference<>();
-
-    private final Context mApplicationContext;
-
+    private final Context mContext;
     private final BroadcastQueryClient mQueryClient;
+    private final AppSettings mAppSettings;
 
     /**
-     * Retrieves the singleton credential client instance associated with the application.
+     * Returns a new credential client instance configured with the default options.
+     *
+     * @See {@link CredentialClientOptions#DEFAULT}.
      */
-    public static CredentialClient getApplicationBoundInstance(Context context) {
-        Context applicationContext = context.getApplicationContext();
-        CredentialClient client = new CredentialClient(applicationContext);
-        if (!INSTANCE_REF.compareAndSet(null, client)) {
-            client = INSTANCE_REF.get();
-        }
+    public static CredentialClient getInstance(@NonNull Context context) {
+        return getInstance(context, CredentialClientOptions.DEFAULT);
+    }
 
-        return client;
+    /**
+     * Returns a new credential client instance configured with the given
+     * {@link CredentialClientOptions}.
+     */
+    public static CredentialClient getInstance(
+            @NonNull Context context,
+            @NonNull CredentialClientOptions options) {
+        return new CredentialClient(
+                context,
+                BroadcastQueryClient.getApplicationBoundInstance(context),
+                options.getAppSettings(context));
     }
 
-    CredentialClient(Context applicationContext) {
-        mApplicationContext = applicationContext;
-        mQueryClient = BroadcastQueryClient.getApplicationBoundInstance(applicationContext);
+    CredentialClient(
+            @NonNull Context context,
+            @NonNull BroadcastQueryClient broadcastQueryClient,
+            @NonNull AppSettings appSettings) {
+        validate(context, notNullValue(), IllegalArgumentException.class);
```

Comment:
The general guidance from "Effective Java" is to throw NullPointerException where the validation check is for nullity, so `require(context, notNullValue(),  NullPointerException.class)` would be better. Same for subsequent validations.